features/steps/argonaut.py

```diff
@@ -344,3 +359,69 @@ def step_impl(context, field_name, value):
                 resource=res,
                 values=found,
                 value=value)
+
+
+def vital_unit_validation(field_name, resource, system_url):
+
+    path = field_name.split('.')
+    path.pop(0)
+
+    systems_to_validate = utils.traverse(resource, path + ["system"])
+    codes_to_validate = utils.traverse(resource, path + ["code"])
+    values_to_validate = utils.traverse(resource, ["component"])
+
+    if not isinstance(systems_to_validate, list):
+        systems_to_validate = [systems_to_validate]
+
+    if not isinstance(codes_to_validate, list):
+        codes_to_validate = [codes_to_validate]
+
+    if not values_to_validate:
```

Comment:
This is redefined pretty far away its initalization; might just avoid the conditional here and definite it initially as the set of (resource + all of its components)?

Reply:
I don't know a prettier pythonic way of conditionally adding those lists if one of them can be null.

Reply:
There's always 

```py
>>> x = None
>>> y = None
>>> (x or []) + (y or [])
[]
```